remote/remote_state.go

```diff
@@ -231,6 +233,36 @@ func (state *RemoteState) GenerateTerraformCode(terragruntOptions *options.Terra
 	// Make sure to strip out terragrunt specific configurations from the config.
 	config := state.Config
 
+	// Initialize the encryption config based on the key provider
+	var encryption map[string]interface{}
+
+	switch {
+	case state.Encryption == nil:
+		terragruntOptions.Logger.Debugf("No encryption block in remote_state config")
+	case len(state.Encryption) == 0:
+		terragruntOptions.Logger.Debugf("Empty encryption block in remote_state config")
+	default:
+		keyProvider, ok := state.Encryption["key_provider"].(string)
```

Comment:
We should reuse the constant.

https://github.com/norman-zon/terragrunt/blob/8c4680593667daa7f2a5d999e7a642a562574f31/codegen/generate.go#L65